src/containers/App.jsx

```diff
@@ -56,59 +70,61 @@ const App = ({ accounts, konnectors, triggers }) => {
   const webviewIntent = useWebviewIntent()
   const theme = useCozyTheme()
 
-  const { shortcutsDirectories } = useCustomShortcuts()
+  const { t } = useI18n()
+
+  const homeMagicFolderConn = mkHomeMagicFolderConn(t)
+  const { data: magicFolder } = useQuery(
+    homeMagicFolderConn.query,
+    homeMagicFolderConn
+  )
+  const magicHomeFolderId = magicFolder?.[0]?._id
+
+  const homeShortcutsDirConn = mkHomeCustomShorcutsDirConn({
+    currentFolderId: magicHomeFolderId
+  })
+  const { data: folders } = useQuery(homeShortcutsDirConn.query, {
+    ...homeShortcutsDirConn.options,
+    enabled: !!magicHomeFolderId
+  })
+  const customHomeShortcutsConn = mkHomeCustomShorcutsConn(
+    folders && folders.map(folder => folder._id)
+  )
+  const { data: customHomeShortcuts } = useQuery(
+    customHomeShortcutsConn.query,
+    {
+      ...customHomeShortcutsConn,
+      enabled: Boolean(folders && folders.length > 0)
+    }
+  )
+
+  const shortcutsDirectories = formatShortcuts(folders, customHomeShortcuts)
+  const context = useQuery(contextQuery.definition, contextQuery.options)
+
   useEffect(() => {
     setIsFetching(
-      [accounts, konnectors, triggers].some(collection =>
+      [accounts, konnectors, triggers, context].some(collection =>
         ['pending', 'loading'].includes(collection.fetchStatus)
       )
     )
     setHasError(
-      [accounts, konnectors, triggers].find(
+      [accounts, konnectors, triggers, context].find(
         collection => collection.fetchStatus === 'failed'
       )
     )
-  }, [accounts, konnectors, triggers])
-  /*  useEffect(() => {
-    client.query(Q('io.cozy.triggers'))
-    client.query(Q('io.cozy.jobs'))
-    client.query(Q('io.cozy.accounts'))
-  }, []) */
-  useEffect(() => {
-    // if we already have the query, let's refresh in "background"
-    // aka without loading state
-    const alreadyRequestedContext = client.getQueryFromState(
-      'io.cozy.settings/io.cozy.settings.context'
-    )
-    if (
-      !alreadyRequestedContext ||
-      alreadyRequestedContext.fetchStatus !== 'loaded'
-    ) {
-      setStatus(FETCHING_CONTEXT)
-    }
-
-    const fetchContext = async () => {
-      const context = await client.query(
-        Q('io.cozy.settings').getById('io.cozy.settings.context')
-      )
-      if (context && context.attributes && context.attributes.features) {
-        const flags = toFlagNames(context.attributes.features)
-        enableFlags(flags)
-      }
-      setStatus(IDLE)
-    }
-    fetchContext()
-  }, [client])
+  }, [accounts, konnectors, triggers, context])
 
+  if (context && context.attributes && context.attributes.features) {
```

Comment:
nit: this is equivalent to `context?.attributes?.features` maybe it would be shorter

Reply:
copy / pasted. Thanks, will update.